src/Veldrid/MTL/MTLGraphicsDevice.cs

```diff
@@ -365,6 +365,8 @@ private protected override void UpdateBufferCore(DeviceBuffer buffer, uint buffe
             var mtlBuffer = Util.AssertSubtype<DeviceBuffer, MTLBuffer>(buffer);
             void* destPtr = mtlBuffer.Pointer;
             byte* destOffsetPtr = (byte*)destPtr + bufferOffsetInBytes;
+
+            Debug.Assert(destPtr != null, "Attempting to write to a MTLBuffer that is inaccessible from a CPU.");
```

Comment:
Shouldn't this throw instead?

Reply:
I would hope there are guards for this at a higher level, but I guess there aren't.